fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java

```diff
@@ -3495,10 +3495,16 @@ private Expression withPredicate(Expression valueExpression, PredicateContext ct
             Expression outExpression;
             switch (ctx.kind.getType()) {
                 case DorisParser.BETWEEN:
-                    outExpression = new And(
-                            new GreaterThanEqual(valueExpression, getExpression(ctx.lower)),
-                            new LessThanEqual(valueExpression, getExpression(ctx.upper))
-                    );
+                    Expression lower = getExpression(ctx.lower);
```

Comment:
support the following case: dt (date type)BETWEEN '2024-08-19' AND       '2024-08-19 14:44:28' =>  BETWEEN '2024-08-19' AND  '2024-08-19' => dt = '2024-08-19'